ExampleMod/Common/GlobalItems/ItemWithGrowingDamage.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Terraria;
+using Terraria.ModLoader;
+using Terraria.ID;
+using Microsoft.Xna.Framework;
+using Terraria.ModLoader.IO;
+
+namespace ExampleMod.Common.GlobalItems
+{
+	internal class ItemWithGrowingDamage : GlobalItem
+	{
+		public int experience;
+		public int level;
+		public static int experiencePerLevel = 100;
+
+		public override bool InstancePerEntity => true;
+
+		public override bool AppliesToEntity(Item entity, bool lateInstantiation) {
+			//Apply this GlobalItem to all swords
+			return entity.DamageType == DamageClass.Melee && entity.noMelee == false && entity.useStyle == ItemUseStyleID.Swing;
+		}
+
+		public override GlobalItem Clone(Item item, Item itemClone) {
+			//Overriding clone is required for GlobalItems that set InstancePerEntity to true.
+			//It is expremely important to include complex fields and properties such as arrays, lists and dictionaries in the Clone method.
+			//If they are not included, they will act like static fields or properties.
+			//It is not required to include simple fields and properties such as int, float, bool.
+			ItemWithGrowingDamage clone = (ItemWithGrowingDamage)base.Clone(item, itemClone);
+			return clone;
+		}
```

Comment:
As much as the comment is nice here, it's not a good example to override a method when you don't need to.

Reply:
I would like to try to keep the comments.  I’ll try to think of a simple way to use a list or dictionary.  Those are things I struggled with a huge amount when creating my GlobalItem class, so I’d like to pass it on to others.  I’ll probably just have it copy the consumed item list and use that somehow.

Reply:
Removed